tests/test_discriminated_union.py

```diff
@@ -2149,6 +2150,7 @@ class Yard(BaseModel):
     assert str(yard_dict['pet']['type']) == 'dog'
 
 
+@pytest.mark.thread_unsafe(reason='Passes on multithreaded. This needs to be investigated further.')
```

Comment:
While all the other tests marked as thread unsafe made sense, I have no idea why this one passes. Core schemas are identical, and verified it isn't a late lambda binding issue. Might something in core?